src/error.js

```diff
@@ -0,0 +1,6 @@
+export default function createEnigmaError(code, name) {
```

Comment:
Could you move the inline comments to a JSDoc formatted comment above the function?

Reply:
Could you also add a spec file for this, that just creates an error with some code / name and verifies it has the correct properties?

Reply:
I'll fix that.... On Monday 😀